### PR TITLE
SmartBFS 2020

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

Do not use other indexes when doing smart breadth first search.
Related Enterprise PR: https://github.com/arangodb/enterprise/pull/491

### Testing & Verification

Test will be added.

### CODE REVIEW

**Comment by maierlars** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/10822/

File: arangod/Aql/OptimizerUtils.cpp [hunk 1]
```diff
@@ -457,19 +457,28 @@ std::pair<bool, bool> findIndexHandleForAndNode(
...
   TRI_ASSERT(node->type == arangodb::aql::AstNodeType::NODE_TYPE_OPERATOR_NARY_AND);
   if (node->numMembers() == 0) {
     // Well no index can serve no condition.
     return false;
   }
-  
+
   auto indexes = collection.indexes();
+  if (onlyEdgeIndexes) {
+    indexes.erase(std::remove_if(indexes.begin(), indexes.end(),
```
Comment (mchacki) [inline arangod/Aql/OptimizerUtils.cpp:476]: This syntax is correct, but weird...

**Comment by maierlars** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/10887/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11050/

**Pull request merged**